src/core/modules/GridLinesMgr.cpp

```diff
@@ -651,6 +658,60 @@ void SkyLine::init()
 		getPrecessionAnglesVondrak(jdY0, &epsilonA, &chiA, &omegaA, &psiA);
 		precessionPartitions.insert(y, psiA); // Store only the value of shift along the ecliptic.
 	}
+	StelCore *core=StelApp::getInstance().getCore();
+	const double jde=core->getJDE();
+	int year, month, day;
+	StelUtils::getDateFromJulianDay(jde, &year, &month, &day);
+	computeEclipticDatePartitions(year);
+}
+
+void SkyLine::computeEclipticDatePartitions(int year)
+{
+	static GridLinesMgr* gMgr=GETSTELMODULE(GridLinesMgr);
+	if (gMgr && (! gMgr->getFlagEclipticDatesLabeled()))
+		return;
+
+	Q_ASSERT(earth);
+	StelCore *core=StelApp::getInstance().getCore();
+	eclipticOnDatePartitions.clear();
+
+	if (year==std::numeric_limits<int>::min())
+	{
+		int m, d;
+		StelUtils::getDateFromJulianDay(core->getJD(), &year, &m, &d);
+	}
+	for (int day=1; day <= (StelUtils::isLeapYear(year) ? 366 : 365); day++)
+	{
+		const Vec3i date=StelUtils::dateFromDayYear(day, year);
+		double jd;
+		StelUtils::getJDFromDate(&jd, year, date[1], date[2], 0, 0, 0.f);
+		const double tzOffset=core->getUTCOffset(jd)/24.;
+		const double jde=jd+core->getDeltaT()/86400.-tzOffset;
+		Vec3d earthPos=earth->getEclipticPos(jde); // J2000 position!
+		double lng, lat, r;
+		StelUtils::rectToSphe(&lng, &lat, &r, earthPos);
+		lng+=M_PI; lat*=-1.; // derive Solar data from this.
+		double epsilonA, chiA, omegaA, psiA;
+		getPrecessionAnglesVondrak(jde, &epsilonA, &chiA, &omegaA, &psiA);
+		lng+=psiA;
+		double aberration=-20.4898/3600.*M_PI_180/r; // Meeus AA (25.10)
+		double deltaEps, deltaPsi;
+		getNutationAngles(jde, &deltaPsi, &deltaEps);
+
+		QString label;
+		if ((year==1582) && (date[1]==10))
+		{
+			// Gregorian calendar reform. Duplicate ticks are harmless, but we must tweak date labels.
+			if (QList<int>({1, 4, 15, 20, 25}).contains(date[2]))
```

Comment:
When a constructor accepts `std::initializer_list`, the outer parentheses can be omitted at the call site (since C++14), resulting in a neater expression: `QList<int>{1, 4, 15, 20, 25}`.

Reply:
Nice!